### PR TITLE
feat(benchmark): add 5 new IT Cloud tests

### PR DESCRIPTION
**Pull request by BonnardValentin**
# 🚀 Add Cloud Infrastructure Benchmark Questions

## Overview
This PR adds five new benchmark questions focused on cloud infrastructure and DevOps practices. These questions are designed to test candidates' practical knowledge of container orchestration, infrastructure as code, load balancing, service discovery, and monitoring - all executable in a local environment.

## Changes
- Added 5 new questions (honest_56 to honest_60) to the benchmark suite
- Created corresponding test scripts for automated validation
- Added necessary template files for each question

### Questions Added:
1. **3-Tier Application (Docker Compose)**
   - Tests understanding of container orchestration
   - Validates service communication and health checks
   - Uses nginx, Flask, and Redis stack

2. **Infrastructure as Code (Terraform)**
   - Tests IaC fundamentals with Docker provider
   - Validates network configuration and container dependencies
   - Includes environment variable management

3. **Load Balancer Configuration (HAProxy)**
   - Tests load balancing concepts
   - Implements health checks and round-robin algorithm
   - Manages multiple backend services

4. **Service Discovery (Consul)**
   - Tests service registration and discovery
   - Implements health monitoring
   - Includes service metadata management

5. **Monitoring Stack (Prometheus/Grafana)**
   - Tests monitoring infrastructure setup
   - Implements metrics collection
   - Includes dashboard configuration

## Related Issues
Issues [ISS-138](https://linear.app/2501ai/issue/ISS-138/510-new-bench-questions-it-cloud) (5  new bench questions IT / Cloud)

### CODE REVIEW

**Review by shide1989 — APPROVED** [review]
lgtm

**Comment by zhuk-aa** [comment]
I guess all of these `/datasets/cloud_xx/` paths were for dev and need to be updated to `honest_xx?`?

**Comment by zhuk-aa** [comment]
Screenshot 2024-12-03 at 13 40 00
would be good to validate if the configs are valid. For example https://developer.hashicorp.com/consul/commands/validate
How hard would it be?

**Comment by BonnardValentin** [comment]
> Screenshot 2024-12-03 at 13 40 00 would be good to validate if the configs are valid. For example https://developer.hashicorp.com/consul/commands/validate How hard would it be?

I will check it

File: config/honest_benchmark.jsonl [hunk 1]
```diff
@@ -53,3 +53,8 @@
...
+{"id": "honest_56", "input": "Create a docker-compose.yml file that sets up a 3-tier application with: nginx as frontend (port 80), a Python Flask API (port 5000), and Redis as backend (port 6379). Make sure the containers can communicate with each other and include proper healthchecks.", "test_script": "import yaml; config = yaml.safe_load(open('./datasets/cloud_1/docker-compose.yml')); output = 'PASS' if len(config['services']) == 3 and all(svc in config['services'] for svc in ['frontend', 'api', 'redis']) else 'FAIL'", "tags": ["docker", "docker-compose", "microservices"]}
```
[the remark this answers is not in the document]
Comment (BonnardValentin) [reply config/honest_benchmark.jsonl:56]: I don't understand ? do you want me to test the docker containers?

**Comment by BonnardValentin** [comment]
I think I've fixed everything

**Review by zhuk-aa — APPROVED** [review]
nice, that looks truly honest 💪

**Pull request merged**